File: IMASampleApp/IMASampleApp/Utils/PlayerSelectionOption.m
/**
 * @class      PlayerSelectionOption PlayerSelectionOption.m "PlayerSelectionOption.m"
 * @brief      An object that contains the information needed to represent an example.
 * @details    An object that contains the information needed to represent an example.
 *             This object is passed between the List and the Player to transfer the information
 * @date       12/12/14
 * @copyright  Copyright (c) 2014 Ooyala, Inc. All rights reserved.
 */

#import "PlayerSelectionOption.h"

@implementation PlayerSelectionOption

- (instancetype)initWithTitle:(NSString *)title
                    embedCode:(NSString *)embedCode
                        pcode:(NSString *)pcode
                       domain:(NSString *)domain
               viewController:(Class)viewController {
  if (self = [super init]) {
    _title = title;
    _embedCode = embedCode;
    _pcode = pcode;
    _domain = domain;
    _viewController = viewController;
  }
  return self;
}

@end

File: PlaybackLab/MultiplePlayersScrollingViewSampleApp/MultiplePlayersScrollingViewSampleApp/Models/PlayerSelectionOption.swift
//
//  PlayerSelectionOption.swift
//  MultiplePlayersScrollingViewSampleApp
//
//  Copyright © 2019 Ooyala. All rights reserved.
//

class PlayerSelectionOption {

  public private(set) var embedCode: String
  public private(set) var pcode: String
  public private(set) var domain: OOPlayerDomain
  public private(set) var embedTokenGenerator: OOEmbedTokenGenerator?
  public var playheadTime: Float64 = 0
  public var isPaused = false
  
  init(embedCode: String,
       pcode: String,
       domain: OOPlayerDomain,
       embedTokenGenerator: OOEmbedTokenGenerator? = nil) {
    self.embedCode = embedCode
    self.pcode     = pcode
    self.domain    = domain
    self.embedTokenGenerator = embedTokenGenerator
  }

}
